lib/ReplaceOpenCLBuiltinPass.cpp

```diff
@@ -903,6 +907,136 @@ bool ReplaceOpenCLBuiltinPass::replaceMadandMad24andMul24(Module &M) {
   return Changed;
 }
 
+bool ReplaceOpenCLBuiltinPass::replaceVstore(Module &M) {
+  bool Changed = false;
+
+  struct VectorStoreOps {
+    const char* name;
+    int n;
+    Type* (*get_scalar_type_function)(LLVMContext&);
```

Comment:
I find this a little baroque.  You've already got M so I would have written this as just Type* in the member declaration, and invoked Type::getFloatTy(M.getContext() in the data.

But no big deal. What you did is fine.  :-)